packages/components/src/internal/AssayDefinitionModel.ts

```diff
@@ -282,19 +284,25 @@ export class AssayDefinitionModel extends Record({
         }
     }
 
-    getDomainColumns(type: AssayDomainTypes): OrderedMap<string, QueryColumn> {
-        let columns = OrderedMap<string, QueryColumn>();
+    getDomainColumns(type: AssayDomainTypes, queryInfo?: QueryInfo): OrderedMap<string, QueryColumn> {
+        const columns = OrderedMap<string, QueryColumn>().asMutable();
 
         if (this.domains && this.domains.size) {
             const domainColumns = this.getDomainByType(type);
 
             if (domainColumns && domainColumns.size) {
                 domainColumns.forEach(dc => {
-                    columns = columns.set(dc.fieldKey.toLowerCase(), dc);
+                    columns.set(dc.fieldKey.toLowerCase(), dc);
                 });
             }
         }
 
-        return columns;
+        // Issue 47576: if there are wrapped columns that are marked as userEditable and shownInInsertView, include them in the form / UI
+        queryInfo?.columns.forEach(c => {
```

Comment:
This hints that getDomainColumns with queryInfo is only to be used for insert/import use case, which is true for now. Still it's a bit weird for a general purpose getDomainColumns to be tied to shownInInsertView prop.

Reply:
Yeah, good point. I will revert this so that getDomainColumns remains as it was and this assay import specific code is in a less general place.